front/src/applications/operationalStudies/helpers/formatTrainScheduleSummaries.ts

```diff
@@ -0,0 +1,84 @@
+import dayjs from 'dayjs';
+import { compact, keyBy } from 'lodash';
+
+import type {
+  LightRollingStockWithLiveries,
+  SimulationSummaryResult,
+  TrainScheduleResult,
+} from 'common/api/osrdEditoastApi';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/TimetableV2/types';
+import { formatToIsoDate, isoDateToMs } from 'utils/date';
+import { jouleToKwh } from 'utils/physics';
+import { formatKmValue } from 'utils/strings';
+import { ISO8601Duration2sec } from 'utils/timeManipulation';
+import type { Dictionary } from 'utils/types';
+
+const getRelevantTrainSchedules = (
+  rawTrainSchedules: Dictionary<TrainScheduleResult>,
+  trainIds: number[]
+) => {
+  const relevantTrainSchedules: TrainScheduleResult[] = [];
+
+  trainIds.forEach((trainId) => {
+    const trainSchedule = rawTrainSchedules[trainId];
+    if (trainSchedule) relevantTrainSchedules.push(trainSchedule);
+  });
+
+  return relevantTrainSchedules;
+};
+
+const formatTrainScheduleSummaries = (
+  trainIds: number[],
+  rawSummaries: Dictionary<SimulationSummaryResult>,
+  rawTrainSchedules: Dictionary<TrainScheduleResult>,
+  rollingStocks: LightRollingStockWithLiveries[]
+): Dictionary<TrainScheduleWithDetails> => {
+  const relevantTrainSchedules = getRelevantTrainSchedules(rawTrainSchedules, trainIds);
+
+  const trainScheduleWithDetails = relevantTrainSchedules.map((trainSchedule) => {
+    const rollingStock = rollingStocks.find((rs) => rs.name === trainSchedule.rolling_stock_name);
+    const trainSummary = rawSummaries[trainSchedule.id];
+
+    if (!trainSummary) return null;
+
+    const otherProps =
+      trainSummary.status === 'success'
+        ? {
+            isValid: true,
+            arrivalTime: formatToIsoDate(
+              isoDateToMs(trainSchedule.start_time) + trainSummary.time,
+              true
+            ),
+            duration: trainSummary.time,
+            pathLength: formatKmValue(trainSummary.length, 'millimeters', 1),
+            mechanicalEnergyConsumed: jouleToKwh(trainSummary.energy_consumption, true),
+            scheduledPointsNotHonored: !trainSummary.scheduled_points_honored,
+          }
+        : {
+            isValid: false,
+            arrivalTime: '',
+            duration: 0,
+            pathLength: '',
+            mechanicalEnergyConsumed: 0,
+            invalidReason: trainSummary.status,
+          };
+
+    return {
+      id: trainSchedule.id,
+      trainName: trainSchedule.train_name,
+      startTime: dayjs(trainSchedule.start_time).format('D/MM/YYYY HH:mm:ss'), // format to time
```

Comment:
this is really generic, can’t this can be in utils/date.ts?

Reply:
This function is not used anywhere else and is going to be removed when we will send date instead of time string to the chart, so I would rather keep it here